riperf3 socket2 dependency is not self-contained

## Summary
When a downstream crate depends on socket2 0.6 and also depends on riperf3 0.6.1, builds can fail in riperf3 with:

`error[E0599]: no method named set_mss found for struct SockRef in the current scope`

This happens because riperf3 uses socket2 0.5 APIs that require the all feature, but riperf3 does not explicitly enable that feature on its own socket2 dependency.

## Environment
- OS: macOS
- Rust: stable
- riperf3: 0.6.1 (published crate)

## Reproduction
Create a minimal crate with this Cargo.toml:

    [package]
    name = "foo"
    version = "0.1.0"
    edition = "2021"

    [workspace]

    [dependencies]
    socket2 = { version = "0.6", features = ["all"] }
    riperf3 = "0.6.1"

Then run:

    cargo check --manifest-path foo/Cargo.toml

## Actual result
Build fails while compiling riperf3 with:

    error[E0599]: no method named `set_mss` found for struct `SockRef<'_>` in the current scope
       --> .../riperf3-0.6.1/src/net.rs:367:26

## Expected result
riperf3 should compile without requiring downstream crates to add a direct socket2 0.5 alias dependency just to enable missing features.

## Root cause
In socket2 0.5, set_mss is gated by feature all. Features are unified per exact crate version. Enabling features on socket2 0.6 does not enable them on socket2 0.5.

## Proposed fix
In riperf3 Cargo.toml, change socket2 dependency to explicitly enable all:

    socket2 = { workspace = true, features = ["all"] }

(or equivalent non-workspace declaration in published crate source)

## Why this fix
- Makes riperf3 self-contained for consumers
- Removes need for downstream workaround like:

    `socket2_05 = { package = "socket2", version = "0.5", features = ["all"] }`

- Preserves current API usage in riperf3 net logic

--- a/riperf3/Cargo.toml
+++ b/riperf3/Cargo.toml
@@ -22,7 +22,7 @@ serde.workspace = true
 sha1.workspace = true
 sha2.workspace = true
 serde_json.workspace = true
-socket2.workspace = true
+socket2 = { workspace = true, features = ["all"] }
 thiserror.workspace = true
 tokio.workspace = true
 

